packages/go_router/lib/src/delegate.dart

```diff
@@ -146,22 +146,38 @@ class GoRouterDelegate extends RouterDelegate<RouteMatchList>
   }
 
   bool _onPopPage(Route<Object?> route, Object? result) {
-    if (!route.didPop(result)) {
```

Comment:
you should call this when https://api.flutter.dev/flutter/widgets/Route/willHandlePopInternally.html is true.
 in this case calling didPop should always return false and the onExit should not be called

Reply:
Sorry I don't understand your this comment. If feel it's kind of in conflict with [your other comment](https://github.com/flutter/packages/pull/3709#discussion_r1179439065).

Here, do you mean I should do something like this?

```dart
if (route.willHandlePopInternally) {
  route.didPop(result);
  return;
}

final Page<Object?> page = route.settings as Page<Object?>;
final RouteMatch? match = builder.getRouteMatchForPage(page);
assert(match != null);
if (match!.route.onExit != null) {
  // ...
}
```

Reply:
correct. although I would add a check to make sure things are working expected
```dart
if (route.willHandlePopInternally) {
  final bool popped = route.didPop(result);
  assert(!popped);
  return popped;
}
```

Reply:
Done in [First check if route will handle pop internally](https://github.com/flutter/packages/pull/3709/commits/b2326d126de4b0676485ba8c024a87bf435a9ead)